framework/OutStreams/PlotEntity.py

```diff
@@ -0,0 +1,128 @@
+import os
+import matplotlib
+
+from utils import utils, InputTypes
+from .OutStreamEntity import OutStreamEntity
+from .PlotInterfaces import factory as interfaceFactory
+
+# initialize display settings
+display = utils.displayAvailable()
+if not display:
+  matplotlib.use('Agg')
+
+
+
+class Plot(OutStreamEntity):
+  """
+    Handler for Plot implementations
+  """
+  @classmethod
+  def getInputSpecification(cls):
+    """
+      Method to get a reference to a class that specifies the input data for class "cls".
+      @ In, cls, the class for which we are retrieving the specification
+      @ Out, inputSpecification, InputData.ParameterInput, class to use for specifying the input of cls.
+    """
+    spec = super().getInputSpecification()
+    okTypes = list(interfaceFactory.knownTypes())
+    okEnum = InputTypes.makeEnumType('OutStreamPlot', 'OutStreamPlotType', okTypes)
+    spec.addParam('subType', required=False, param_type=okEnum, descr=r"""Type of OutStream Plot to generate.""")
+    # TODO add specs depending on the one chosen, not all of them!
+    # FIXME the GeneralPlot has a vast need for converting to input specs. Until then,
+    #       we cannot strictly check anything related to it.
+    spec.strictMode = False
+    for name in okTypes:
+      plotter = interfaceFactory.returnClass(name)
+      subSpecs = plotter.getInputSpecification()
+      spec.mergeSub(subSpecs)
+    return spec
+
+  def __init__(self):
+    """
+      Init of Base class
+      @ In, None
+      @ Out, None
+    """
+    super().__init__()
+    self.printTag = 'PlotEntity'
+    self._plotter = None            # implemention, inheriting from interface
+
+  def _readMoreXML(self, xml):
+    """
+      Legacy passthrough.
+      @ In, xml, xml.etree.ElementTree.Element, input
+      @ Out, None
+    """
+    # TODO remove this when GeneralPlot conforms to inputParams
+    subType = xml.attrib.get('subType', 'GeneralPlot').strip()
+    if subType == 'GeneralPlot':
+      self._plotter = interfaceFactory.returnInstance(subType)
+      self._plotter.handleInput(xml)
+    else:
+      spec = self.parseXML(xml)
+      self.handleInput(spec)
+
+  def _handleInput(self, spec):
+    """
+      Loads the input specs for this object.
+      @ In, spec, InputData.ParameterInput, input specifications
+      @ Out, None
+    """
+    super()._handleInput(spec)
+    # we specialized out the GeneralPlot in _readMoreXML, so here we have a real inputParams user
+    reqType = spec.parameterValues['subType']
+    self._plotter = interfaceFactory.returnInstance(reqType)
+    self._plotter.handleInput(spec)
+
+  def initialize(self, stepEntities):
+    """
+      Initialize the OutStream. Initialize interfaces and pass references to sources.
+      @ In, stepEntities, dict, the Entities used in the current Step. Sources are taken from this.
+      @ Out, None
+    """
+    super().initialize(stepEntities)
+    self._plotter.initialize(stepEntities)
+
+  def addOutput(self):
```

Comment:
Is it required method? Should this method be abstract class in the OutStreamEntity class?

Reply:
we could, yes; making it abstract will prevent us from implementing any common code for all the `OutStreams` to be run during `addOutput`. I don't see this as desirable, but am happy to make the change if it is, @alfoa

Reply:
okay, added